src/main/java/com/google/sps/ColorUtils.java

```diff
@@ -0,0 +1,136 @@
+    colorList.add(new ColorName("Pink", 0xFF, 0x69, 0xB4));       // HotPink
+    colorList.add(new ColorName("Indigo", 0x4B, 0x00, 0x82));
+    colorList.add(new ColorName("Khaki", 0xF0, 0xE6, 0x8C));
+    colorList.add(new ColorName("Lavender", 0xE6, 0xE6, 0xFA));
+    colorList.add(new ColorName("Light Blue", 0xAD, 0xD8, 0xE6));
+    colorList.add(new ColorName("Light Cyan", 0xE0, 0xFF, 0xFF));
+    colorList.add(new ColorName("Light Gray", 0xD3, 0xD3, 0xD3));
+    colorList.add(new ColorName("Light Green", 0x90, 0xEE, 0x90));
+    colorList.add(new ColorName("Light Pink", 0xFF, 0xB6, 0xC1));
+    colorList.add(new ColorName("LightYellow", 0xFF, 0xFF, 0xE0));
+    colorList.add(new ColorName("Lime", 0x00, 0xFF, 0x00));
+    colorList.add(new ColorName("Magenta", 0xFF, 0x00, 0xFF));
+    colorList.add(new ColorName("Maroon", 0x80, 0x00, 0x00));
+    colorList.add(new ColorName("Blue", 0x00, 0x00, 0xCD));       // MediumBlue
+    colorList.add(new ColorName("Purple", 0x93, 0x70, 0xDB));     // MediumPurple
+    colorList.add(new ColorName("Green", 0x00, 0xFA, 0x9A));      // MediumSpringGreen
+    colorList.add(new ColorName("Turquoise", 0x48, 0xD1, 0xCC));  // MediumTurquoise
+    colorList.add(new ColorName("Navy", 0x00, 0x00, 0x80));
+    colorList.add(new ColorName("Orange", 0xFF, 0xA5, 0x00));
+    colorList.add(new ColorName("Orange", 0xFF, 0x45, 0x00));     // OrangeRed
+    colorList.add(new ColorName("Pink", 0xFF, 0xC0, 0xCB));
+    colorList.add(new ColorName("Purple", 0x80, 0x00, 0x80));
+    colorList.add(new ColorName("Red", 0xFF, 0x00, 0x00));
+    colorList.add(new ColorName("Blue", 0x41, 0x69, 0xE1));       // RoyalBlue
+    colorList.add(new ColorName("Brown", 0x8B, 0x45, 0x13));      // SaddleBrown
+    colorList.add(new ColorName("Silver", 0xC0, 0xC0, 0xC0));
+    colorList.add(new ColorName("Blue", 0x87, 0xCE, 0xEB));       // SkyBlue
+    colorList.add(new ColorName("Turquoise", 0x40, 0xE0, 0xD0));
+    colorList.add(new ColorName("Violet", 0xEE, 0x82, 0xEE));
+    colorList.add(new ColorName("White", 0xFF, 0xFF, 0xFF));
+    colorList.add(new ColorName("Yellow", 0xFF, 0xFF, 0x00));
+    return colorList;
+  }
+  
+  /**
+   * Returns the color name of the closest color (to the query color provided as RGB).
+   * To find the closest color, gets the color list, and then computes the distance between 
+   * the query color and every color in the color list, based on their RGB values.
+   */
+  public String getColorNameFromRGB(int r, int g, int b) {
+    ArrayList<ColorName> colorList = initColorList();
+    
+    // Get the color name of the color with minimum RGB distance to the query color.
+    ColorName closestColorMatch = null;
+
+    int minRGBDistance = Integer.MAX_VALUE;
+    for (ColorName color : colorList) {
+      int currentRGBDistance = color.computeRGBDistance(r, g, b);
+
+      if (currentRGBDistance < minRGBDistance) {
+        minRGBDistance = currentRGBDistance;
+        closestColorMatch = color;
```

Comment:
Yes, so I'm calculating the distance between the query color and each color in the colorList, as you said, and from these values I need to select the minimum - which cannot be solved in less than O(N). Calculating the distance is just a formula so O(1). There's no need for hashing here :)

Reply:
I was suggesting something a bit different from what you are saying, maybe I didn't put it correctly.
What I was thinking is you put different colours in different buckets or packets like you assign a bucket/packet for all shades of red, then a different bucket/package for all shades of black and assign them codes/numbers for their rgb range.
When the CloudVision gives you back the rgb of the product you check which range it is in then send it to be checked in that range's bucket. Which will reduce our distance calculation time which we are currently doing for every shade of colour known.

Reply:
Your idea is very interesting! However, would the efficiency improvement be worth it? Now, there are 50 colors in the list and I plan to update it (as said previously), so there will be less than 50 in the end, so that makes the function run in less than ~O(50), which is already effecient. Also, if I follow this buckets approach, how will I choose where to put colors that are "at limit". For instance, for the red group, what if I have a more violet-red which is in there, but that color has a smaller difference to the blue group?

Reply:
There are efficient algorithms to find nearest neighbors in Euclidean space, but I doubt that any will outperform a linear search for a static set of 50 items.

Reply:
Yes, I understand, this idea won't make the complexity any better.